SMP/src/train_amp.py

```diff
@@ -0,0 +1,302 @@
+                    f'Epoch [{epoch+1}/{NUM_EPOCHS}], '
+                    f'Step [{step+1}/{len(train_loader)}], '
+                    f'Loss: {round(loss.item(),4)}'
+                )
+        
+        wandb.log({"train/loss": loss.item()}, step=epoch)
+
+        # validation 주기에 따라 loss를 출력하고 best model을 저장합니다.
+        if (epoch + 1) % VAL_EVERY == 0:
+            dice = validation(epoch + 1, model, val_loader, criterion)
+            
+            if best_dice < dice:
+                print(f"Best performance at epoch: {epoch + 1}, {best_dice:.4f} -> {dice:.4f}")
+                print(f"Save model in {SAVED_DIR}")
+                best_dice = dice
+                save_model(model)
+        
+        if scheduler:
+            if is_plateau:
+                if (epoch + 1) % VAL_EVERY == 0:
+                    scheduler.step(dice)
+            else:
+                scheduler.step()
+
+
+if __name__ == '__main__':
+    with open('config.yaml', 'r') as file:
+        config = yaml.safe_load(file)
+
+    DATA_ROOT = config['DATA_ROOT']
+    IMAGE_ROOT = f"{DATA_ROOT}/train/DCM"
+    LABEL_ROOT = f"{DATA_ROOT}/train/outputs_json"
+    SAVED_DIR = config['SAVED_DIR']
+    EXP_NAME = config['EXP_NAME']
+    RESULT_DIR = os.path.join(SAVED_DIR, EXP_NAME)
+
+    if not os.path.exists(RESULT_DIR):                                                           
+        os.makedirs(RESULT_DIR)
+
+    BATCH_SIZE = config['BATCH_SIZE']
+    LR = config['LR']
+    RANDOM_SEED = config['RANDOM_SEED']
+    NUM_EPOCHS = config['NUM_EPOCHS']
+    VAL_EVERY = config['VAL_EVERY']
+    PSUEDOLABEL_FLAG = config['PSEUDO_LABEL']
+
+    # model 정의
+    TYPE = config['TYPE']
+    MODEL = config['MODEL']
+    ENCODER = config['ENCODER']
+    RESIZE = config['RESIZE']
+    
+    clear_test_data_in_train_path(DATA_ROOT)
+    if PSUEDOLABEL_FLAG:
+        preprocess(DATA_ROOT, config['OUTPUT_CSV_PATH'])
+
+    augmentation_config = config['augmentation']
+    augmentation_name = augmentation_config['name']
+    augmentation_params = augmentation_config['params'] or {}
+    train_tf = create_augmentation(augmentation_name, resize=RESIZE, **augmentation_params)
+    valid_tf = create_augmentation('base', resize=RESIZE)
+
+    train_dataset = XRayDataset(
+        IMAGE_ROOT, 
+        LABEL_ROOT, 
+        is_train=True, 
+        transforms=train_tf,
+        psuedo_flag=PSUEDOLABEL_FLAG,
+    )
+    valid_dataset = XRayDataset(
+        IMAGE_ROOT, 
+        LABEL_ROOT, 
+        is_train=False, 
+        transforms=valid_tf,
+    )
+    
+    if PSUEDOLABEL_FLAG:
+        copy_test_data_to_train_path("data")
+
+    print(len(train_dataset), len(valid_dataset))
+
+    train_loader = DataLoader(
+        dataset=train_dataset, 
+        batch_size=BATCH_SIZE,
+        shuffle=True,
+        num_workers=0,
+        drop_last=True,
+    )
+
+    # 주의: validation data는 이미지 크기가 크기 때문에 `num_wokers`는 커지면 메모리 에러가 발생할 수 있습니다.
```

Comment:
Nice warning. 

What about printing this in the beginning of the program (for unfamiliar users)?